apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/*global Compose, Recipients, Utils, Template, Settings,
+         SMIL, ErrorDialog, MessageManager, LinkHelper,
+         ActivityPicker, InboxView, OptionMenu, Threads, Contacts,
+         Attachment, WaitingScreen, MozActivity, LinkActionHandler,
+         TimeHeaders, ContactRenderer, Draft, Drafts,
+         MultiSimActionButton, Navigation, Promise, LazyLoader,
+         App,
+         SharedComponents,
+         ActivityClient,
+         Errors,
+         EventDispatcher,
+         SelectionHandler,
+         TaskRunner
+*/
+/*exported ConversationView */
+
+(function(exports) {
+'use strict';
+
+var attachmentMap = new WeakMap();
+
+function conv_mmsAttachmentClick(target) {
+  var attachment = attachmentMap.get(target);
+  if (!attachment) {
+    return false;
+  }
+
+  attachment.view({
+    allowSave: true
+  });
+
+  return true;
+}
+
+// reduce the Composer.getContent() into slide format used by SMIL.generate some
+// day in the future, we should make the SMIL and Compose use the same format
+function conv_generateSmilSlides(slides, content) {
+  var length = slides.length;
+  if (typeof content === 'string') {
+    if (!length || slides[length - 1].text) {
+      slides.push({
+        text: content
+      });
+    } else {
+      slides[length - 1].text = content;
+    }
+  } else {
+    slides.push({
+      blob: content.blob,
+      name: content.name
+    });
+  }
+  return slides;
+}
+
+var ConversationView = {
+  CHUNK_SIZE: 10,
+  // duration of the notification that message type was converted
+  CONVERTED_MESSAGE_DURATION: 3000,
```

Comment:
Well, the problem is that we have a bunch of supporting code that I don't really know how to generalize. For example:

1. When subject limit is reached - we show banner that will go away after timeout, but we should hide it as soon as length doesn't exceed limit anymore - so here we should keep reference to "notification" to be able to hide it;

2. When message is resized we show "persistent" notification that should be removed (in our current logic we don't hide it immediately like in example#1, but with with timeout :() only when resize is finished. Also when we said that notification should be hidden in two seconds, it may happen that resizing is happening again, so that we should stop "hiding" :)

So as you see it has quite a bit code that either should be duplicated in both views or moved to Compose. What do you think?

Btw, I have a draft of ```AppNotification``` (still don't like some pieces of it) that can cover all our cases I think: https://github.com/mozilla-b2g/gaia/pull/30794/